agent-rpm-x86_64 package is failing due to the checkfiles test for 4.14.8

## Description

While building the Wazuh agent packages for **4.14.8** (branch `4.X`), the `agent-rpm-x86_64` package generation workflow failed in the **Check installed files** stage (`checkfiles` test).

The `Compare results` step of the `check_files.py` validation reported a mismatch between the installed files and the expected baseline:

```console
Failed: Results didn't match the expected output
Wazuh File Integrity Check
Scanning started...
Scanning finished
##[error]Process completed with exit code 1.
```

The command that failed (size check enabled):

```console
venv/bin/python3 check_files.py --report '/tmp/report.md' --file_csv_path 'rpm_linux_agent_amd64.csv' --wazuh_uid 499 --wazuh_gid 498 --directory '/var/ossec' --size_check
```

All previous stages (build, install, uninstall, upgrade from 4.14.7 to 4.14.8, S3 uploads) completed successfully; only the `Compare results` step failed.

> [!NOTE]
> The detailed per-file diff is written to the job **Summary** (`report.md`) — artifact uploads were disabled in this run (`Patch check_files - skip artifact uploads`), so the report is only available from the run summary linked below and is not attached as an artifact.

## Context

- **Version:** 4.14.8 (branch `4.X`)
- **Package:** `wazuh-agent-4.14.8-wqa8260.x86_64.rpm`
- **Job:** `Build rpm wazuh-agent on x86_64`
- **Failing stage:** `Check installed files` → `Compare results`
- **Trigger:** `workflow_dispatch`

## Evidence

- Failed run: https://github.com/wazuh/wazuh-agent-packages/actions/runs/31080638163/job/92548426931

--- a/.github/actions/check_files/check_files.py
+++ b/.github/actions/check_files/check_files.py
@@ -258,6 +258,31 @@ def get_current_items(scan_path='/var/ossec', size_check=False, ignore_names=[])
 """
 
 
+def path_match(filename, pattern):
+    """Match a filename against a glob pattern segment by segment.
+
+    A wildcard is not allowed to span path separators: both the filename
+    and the pattern are split into their components and each pair is
+    matched independently. This keeps a '*' from crossing a directory
+    boundary regardless of how many characters it expands to. Both '/'
+    (Unix) and '\\' (Windows) separators are supported.
+
+    Args:
+        filename (str): The current file path being checked.
+        pattern (str): The expected path, possibly containing glob wildcards.
+
+    Returns:
+        bool: True if the filename matches the pattern.
+    """
+    file_parts = filename.replace('\\', '/').split('/')
+    pattern_parts = pattern.replace('\\', '/').split('/')
+
+    if len(file_parts) != len(pattern_parts):
+        return False
+
+    return all(fnmatch.fnmatch(f, p) for f, p in zip(file_parts, pattern_parts))
+
+
 def csv_to_dict(file_path, key_column):
     data_dict = {}
 
@@ -466,9 +491,10 @@ if __name__ == "__main__":
             # 2. Check for matches with glob patterns if no exact match found
             if not matched:
                 for pattern, expected_item_fields in glob_patterns.items():
-                    # Check if the pattern matches the current file name
-                    # Note: Only the current directory is considered (no subdirectories)
-                    if fnmatch.fnmatch(current_file_name, pattern) and '/' not in current_file_name[len(pattern)-1:]:
+                    # Check if the pattern matches the current file name.
+                    # Matching is done per path segment so a wildcard never
+                    # spans a directory separator.
+                    if path_match(current_file_name, pattern):
                         matched = True
                         matches[pattern] = True
 

--- a/.github/actions/check_files/test_check_files.py
+++ b/.github/actions/check_files/test_check_files.py
@@ -1,0 +1,60 @@
+import sys
+import types
+
+import pytest
+
+# check_files.py imports pandas at module load time, but path_match does not
+# need it. Provide a lightweight stub so the module can be imported in a plain
+# test environment (the real pandas is used if it happens to be installed).
+sys.modules.setdefault('pandas', types.ModuleType('pandas'))
+
+from check_files import path_match  # noqa: E402
+
+
+class TestPathMatch:
+    """Tests for the glob matching used against the expected-files CSV."""
+
+    @pytest.mark.parametrize("filename", [
+        # The temp directory name expands to different lengths; every one of
+        # these must match the same pattern. Long expansions used to break the
+        # previous length-based check and were wrongly reported as unexpected.
+        "/var/ossec/tmp/sca-4.9-x-tmp/rhel/8/sca.files",
+        "/var/ossec/tmp/sca-4.14.8-a1b2c3-tmp/rhel/8/sca.files",
+        "/var/ossec/tmp/sca-4.14.8-12345678901234-tmp/rhel/8/sca.files",
+        "/var/ossec/tmp/sca-4.14.8-0000000000000000000000-tmp/rhel/8/sca.files",
+    ])
+    def test_wildcard_in_middle_segment_any_length(self, filename):
+        pattern = "/var/ossec/tmp/sca-4.*-*-tmp/rhel/8/sca.files"
+        assert path_match(filename, pattern)
+
+    def test_wildcard_does_not_cross_directory_separator(self):
+        # A '*' inside a segment must not swallow additional directories.
+        pattern = "/var/ossec/tmp/sca-4.*-*-tmp/rhel/8/sca.files"
+        filename = "/var/ossec/tmp/sca-4.14.8-x/extra/y-tmp/rhel/8/sca.files"
+        assert not path_match(filename, pattern)
+
+    def test_trailing_wildcard_matches_exact_and_suffix(self):
+        pattern = "/var/ossec/ruleset/sca/cis_alma_linux_8.yml*"
+        assert path_match("/var/ossec/ruleset/sca/cis_alma_linux_8.yml", pattern)
+        assert path_match("/var/ossec/ruleset/sca/cis_alma_linux_8.yml.gz", pattern)
+
+    def test_trailing_wildcard_does_not_match_deeper_path(self):
+        pattern = "/var/ossec/ruleset/sca/cis_alma_linux_8.yml*"
+        filename = "/var/ossec/ruleset/sca/cis_alma_linux_8.yml/child"
+        assert not path_match(filename, pattern)
+
+    def test_different_segment_count_is_not_a_match(self):
+        pattern = "/var/ossec/tmp/sca-4.*-*-tmp/rhel/8/sca.files"
+        assert not path_match("/var/ossec/tmp/sca-4.14.8-x-tmp/rhel/8", pattern)
+
+    def test_literal_path_without_wildcards(self):
+        pattern = "/var/ossec/etc/ossec.conf"
+        assert path_match("/var/ossec/etc/ossec.conf", pattern)
+        assert not path_match("/var/ossec/etc/ossec.conf.bak", pattern)
+
+    def test_windows_backslash_paths(self):
+        pattern = r"C:\Program Files (x86)\ossec-agent\*.dll"
+        assert path_match(r"C:\Program Files (x86)\ossec-agent\dbsync.dll", pattern)
+        # The wildcard must stay within its segment on Windows too.
+        assert not path_match(
+            r"C:\Program Files (x86)\ossec-agent\sub\dbsync.dll", pattern)

